src/components/_app/Web3ConnectionManager/components/WalletSelectorModal/components/GoogleLoginButton/hooks/useLoginWithGoogle.ts

```diff
@@ -152,24 +153,43 @@ const useLoginWithGoogle = () => {
         address: walletClient.account.address,
       }
 
+      let userId: number
+
       const keyPair = !!addressLinkParams?.userId
         ? await onLinkAddress({ ...addressLinkParams, signProps }).then(
             (result) => result?.keyPair
           )
         : await onSetKeypairSubmit({
             signProps,
-          }).then((result) => result?.keyPair)
+          }).then((result) => {
+            if (result?.user) {
+              userId = result.user?.id
+            }
+            return result?.keyPair
+          })
 
       // 4) Try to connect Google account
-      await onConnectGoogleSubmit({
+      const platformUser = await onConnectGoogleSubmit({
         signOptions: {
           keyPair: keyPair.keyPair,
           ...signProps,
         },
         platformName: "GOOGLE",
         authData,
+        disconnectFromExistingUser: true,
       })
 
+      if (userId && platformUser) {
+        await mutate(
+          [`/v2/users/${userId}/profile`, { method: "GET", body: {} }],
```

Comment:
Totally irrelevant for this PR, but I wonder if we should create a function which generates this SWR key? I noticed that we use it in multiple places, maybe it would make sense to store it in one place, so in case we change it in the future, we won't break anything.

Reply:
Hmm, yeah, that might be worth implementing later on! 💯